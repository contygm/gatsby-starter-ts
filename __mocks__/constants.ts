const glossaryDefinitions: Array<GlossaryElements> = [
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Air',
            letter: 'A',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test'],
            relatedPosts: ['/blog/test1', '/wiki/test1'],
            tags: ['two', 'three']
        }
    },
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Atom',
            letter: 'A',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test'],
            relatedPosts: ['/blog/test1', '/wiki/test1'],
            tags: ['two', 'one']
        }
    },
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Cloud',
            letter: 'C',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test'],
            relatedPosts: ['/blog/test1', '/wiki/test1'],
            tags: ['two', 'one']
        }
    },
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Moon',
            letter: 'M',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test'],
            relatedPosts: ['/blog/test1', '/wiki/test1'],
            tags: ['two']
        }
    },
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Window',
            letter: 'W',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test1'],
            relatedPosts: ['/wiki/test', '/blog/test1'],
            tags: ['three']
        }
    }
];

const featuredOrRelatedPosts: Array<IndexElements> = [
    {
        excerpt:
            'Lemon drops pastry danish halvah. Chupa chups fruitcake powder...',
        fields: {
            slug: '/baking-tutorial/'
        },
        frontmatter: {
            title: 'Baking Tutorial',
            date: '3-2-2021',
            description: 'a baking tutorial',
            tags: ['two', 'three'],
            headerImage: {
                childImageSharp: {
                    gatsbyImageData: {}
                }
            }
        }
    },
    {
        excerpt:
            'Cheesecake lemon drops Cheesecake lemon drops pastry danish halvah...',
        fields: {
            slug: '/ten-best-things/'
        },
        frontmatter: {
            title: '10 Best Things',
            description: 'a list of the ten best things',
            tags: ['one', 'two', 'three'],
            date: '2-2-2021',
            headerImage: {
                childImageSharp: {
                    gatsbyImageData: {}
                }
            }
        }
    },
    {
        excerpt: 'Chupa chups fruitcake powder dessert gingerbread gummies...',
        fields: {
            slug: '/guide/'
        },
        frontmatter: {
            title: 'Guide',
            description: 'a guide',
            tags: ['one', 'two'],
            date: '5-2-2021',
            headerImage: {
                childImageSharp: {
                    gatsbyImageData: {}
                }
            }
        }
    }
];

const previous = {
    fields: {
        slug: '/post-before'
    },
    frontmatter: {
        title: 'Before - A Post About Things'
    }
};

const next = {
    fields: {
        slug: '/post-after'
    },
    frontmatter: {
        title: 'After - A Post About Things'
    }
};

const siteData: SiteMetadata = {
    title: `Starter`,
    siteUrl: `https://www.domain.com`,
    author: {
        name: `Writer B. Author`,
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla quam velit, vulputate eu pharetra nec, mattis ac neque. Duis vulputate commodo lectus, ac blandit elit tincidunt id. Sed rhoncus, tortor sed eleifend tristique, tortor mauris molestie elit'
    },
    businessName: 'The Company',
    phone: '999-999-9999',
    email: 'email@email.com',
    description: 'a super duper site!',
    image: '/nothing.png'
};

export {
    featuredOrRelatedPosts,
    previous,
    next,
    siteData,
    glossaryDefinitions
};
