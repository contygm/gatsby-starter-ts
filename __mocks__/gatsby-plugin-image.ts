import React from 'react';
const plugin = jest.requireActual('gatsby-plugin-image');

const mockImage = ({ imgClassName, ...props } : {imgClassName: string }) =>
    React.createElement('img', {
        ...props,
        className: imgClassName
    });

module.exports = {
    ...plugin,
    GatsbyImage: jest.fn().mockImplementation(mockImage),
    StaticImage: jest.fn().mockImplementation(mockImage)
};
