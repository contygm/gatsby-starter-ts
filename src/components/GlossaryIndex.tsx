import React, { FunctionComponent } from 'react';
import DefinitionCard from './DefinitionCard';

interface GlossaryIndexProps {
    allDefinitions: Array<GlossaryElements>;
    // handleFilterUpdate: (e: any) => void;
}

const GlossaryIndex: FunctionComponent<GlossaryIndexProps> = ({
    allDefinitions
}: // handleFilterUpdate
GlossaryIndexProps) => {
    return (
        <article className="section">
            <section className="container is-max-desktop">
                {/* post cards */}
                <div className="columns is-multiline is-centered">
                    {allDefinitions.length > 0 ? (
                        allDefinitions.map((def: GlossaryElements) => {
                            return (
                                <div
                                    data-testid={'post-card'}
                                    className="column is-4"
                                    key={def.frontmatter.title.toLowerCase()}
                                >
                                    <DefinitionCard
                                        definition={def}
                                        // handleFilterUpdate={handleFilterUpdate}
                                    />
                                </div>
                            );
                        })
                    ) : (
                        // TODO make no results look like something
                        <div>No results</div>
                    )}
                </div>
            </section>
        </article>
    );
};

export default GlossaryIndex;
