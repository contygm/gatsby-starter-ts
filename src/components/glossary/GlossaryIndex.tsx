import React, { FunctionComponent } from 'react';
import { DefinitionCard } from './DefinitionCard';

export interface GlossaryIndexProps {
    allDefinitions: Array<GlossaryElements>;
    allLetters?: {
        group: Array<{
            fieldValue: string;
        }>;
    };
    // handleFilterUpdate: (e: any) => void;
}

const letterDefinitionObjectArray = (
    allLetters: GlossaryIndexProps['allLetters'],
    allDefinitions: GlossaryIndexProps['allDefinitions']
) => {
    let index = 0;
    const resArray: any[] = [];
    allLetters?.group.forEach((letterObj) => {
        const newLetterDefObj = {
            letter: letterObj.fieldValue,
            definitions: [] as Array<GlossaryElements>
        };

        while (
            index < allDefinitions.length &&
            letterObj.fieldValue === allDefinitions[index].frontmatter.letter
        ) {
            newLetterDefObj.definitions.push(allDefinitions[index]);
            index++;
        }
        resArray.push(newLetterDefObj);
    });
    return resArray;
};

export const GlossaryIndex: FunctionComponent<GlossaryIndexProps> = ({
    allDefinitions,
    allLetters
}: // handleFilterUpdate
GlossaryIndexProps) => {
    const letterDefinitions = letterDefinitionObjectArray(
        allLetters,
        allDefinitions
    );

    return (
        <article className="container">
            {/* post cards */}
            <div className="is-centered">
                {letterDefinitions.length > 0 ? (
                    letterDefinitions.map((defObj) => {
                        if (defObj.definitions.length > 0) {
                            return (
                                <section
                                    className="section pt-0"
                                    key={defObj.letter}
                                >
                                    <h2
                                        className="title is-size-2"
                                        id={`${defObj.letter}`}
                                    >
                                        {defObj.letter}
                                    </h2>
                                    {defObj.definitions.map(
                                        (definition: GlossaryElements) => {
                                            return (
                                                <div
                                                    data-testid={
                                                        'definition-card'
                                                    }
                                                    className=""
                                                    key={definition.frontmatter.title.toLowerCase()}
                                                >
                                                    <DefinitionCard
                                                        definition={definition}
                                                        includeAll={true}
                                                        // handleFilterUpdate={handleFilterUpdate}
                                                    />
                                                </div>
                                            );
                                        }
                                    )}
                                </section>
                            );
                        }
                    })
                ) : (
                    <h2 className="title is-size-3">No results</h2>
                )}
            </div>
        </article>
    );
};

// export default GlossaryIndex;
