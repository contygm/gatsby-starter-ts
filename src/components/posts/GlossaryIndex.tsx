import React from 'react';
import { makeLetterDefinitions } from '../../utils/helpers/makeLetterDefinitions';
import { DefinitionCard } from '../common/DefinitionCard';

/**
 * Props for the Glossary Index component
 * @property {GlossaryElements[]} allDefinitions - all glossary definitions
 * @property {Object} [allLetters] - nesting for letters
 * @property {Array<{fieldValue: string}>} allLetters.group - all starting letters using in the glossary in alphabetical order
 *
 * @see GlossaryElements
 * @see GlossaryIndex
 * @category Components
 */
export interface GlossaryIndexProps {
    allDefinitions: GlossaryElements[];
    allLetters?: {
        group: {
            fieldValue: string;
        }[];
    };
}

/**
 * Props for the Glossary Index component
 * @property {GlossaryIndexProps} props - includes all available definitions and all letters used
 *
 * @category Components
 * @see GlossaryIndexProps
 */
export const GlossaryIndex = (props: GlossaryIndexProps) => {
    const letterDefinitions = makeLetterDefinitions(
        props.allLetters,
        props.allDefinitions
    );

    return (
        <article className="container">
            {/* post cards */}
            <div className="is-centered">
                {letterDefinitions.length > 0 ? (
                    letterDefinitions.map((defObj) => {
                        if (defObj.definitions.length > 0) {
                            return (
                                <section
                                    className="gloss-section-wrapper"
                                    key={defObj.letter}
                                >
                                    <h2
                                        className="title-two"
                                        id={`${defObj.letter}`}
                                    >
                                        {defObj.letter}
                                    </h2>
                                    {defObj.definitions.map(
                                        (definition: GlossaryElements) => {
                                            return (
                                                <div
                                                    data-testid="definition-card"
                                                    key={definition.frontmatter.title.toLowerCase()}
                                                >
                                                    <DefinitionCard
                                                        definition={definition}
                                                        includeAll={true}
                                                    />
                                                </div>
                                            );
                                        }
                                    )}
                                </section>
                            );
                        }
                    })
                ) : (
                    <h2 className="title-three">No results</h2>
                )}
            </div>
        </article>
    );
};
