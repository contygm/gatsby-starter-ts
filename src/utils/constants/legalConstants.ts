/**
 * All the legal page information as a constant. Object format is the same as the legal link props
 */
export const legalPageMap = [
    {
        title: 'Privacy Policy',
        cyTag: 'legal-index-privacy',
        url: '/legal/privacy-policy',
        description:
            'Bonbon tootsie roll lemon drops topping biscuit cookie. Icing gummi bears ice cream marshmallow jelly brownie icing soufflé.'
    },
    {
        title: 'Copyright',
        cyTag: 'legal-index-copyright',
        url: '/legal/copyright',
        description:
            'Bonbon tootsie roll lemon drops topping biscuit cookie. Icing gummi bears ice cream marshmallow jelly brownie icing soufflé.'
    },
    {
        title: 'Cookie Policy',
        cyTag: 'legal-index-cookie',
        url: '/legal/cookie-policy',
        description:
            'Bonbon tootsie roll lemon drops topping biscuit cookie. Icing gummi bears ice cream marshmallow jelly brownie icing soufflé.'
    },
    {
        title: 'Terms and Conditions',
        cyTag: 'legal-index-terms',
        url: '/legal/terms-and-conditions',
        description:
            'Bonbon tootsie roll lemon drops topping biscuit cookie. Icing gummi bears ice cream marshmallow jelly brownie icing soufflé.'
    },
    {
        title: 'Disclaimer',
        cyTag: 'legal-index-disclaimer',
        url: '/legal/disclaimer',
        description:
            'Bonbon tootsie roll lemon drops topping biscuit cookie. Icing gummi bears ice cream marshmallow jelly brownie icing soufflé.'
    }
];
