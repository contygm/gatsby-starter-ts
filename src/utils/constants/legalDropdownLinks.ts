/**
 * All the legal page links with labels
 * @property {string} label - The display label for the page
 * @property {string} slug - The slug-friendly version of the label. Should be lowercase and replace spaces with `-`
 * 
 * @memberof Footer
 * @see DropDownLinkBtn
 * @constant {Object[]}
 */
export const LEGAL_LINKS = [
    {
        label: 'Copyright',
        slug: 'copyright'
    },
    {
        label: 'Cookie Policy',
        slug: 'cookie-policy'
    },
    {
        label: 'Terms and Conditions',
        slug: 'terms-and-conditions'
    },
    {
        label: 'Disclaimer',
        slug: 'disclaimer'
    }
];