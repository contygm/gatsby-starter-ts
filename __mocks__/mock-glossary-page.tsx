/* istanbul ignore file */
import { GlossaryPageProps } from '../src/pages/glossary';
import { mockHeadProps, mockPageProps } from '../src/utils/test-props';
import { HeadProps, PageProps } from 'gatsby';
import {
    glossaryDefinitions,
    featuredOrRelatedPosts,
    siteData
} from './constants';

export const mockGlossary: GlossaryPageProps = {
    site: siteData,
    allTags: {
        group: [
            { fieldValue: 'three', totalCount: 3 },
            { fieldValue: 'two', totalCount: 4 },
            { fieldValue: 'one', totalCount: 2 }
        ]
    },
    allLetters: {
        group: [
            { fieldValue: 'A' },
            { fieldValue: 'C' },
            { fieldValue: 'M' },
            { fieldValue: 'W' }
        ]
    },
    index: {
        totalCount: 5,
        nodes: glossaryDefinitions
    },
    blogFeatured: {
        nodes: featuredOrRelatedPosts
    },
    wikiFeatured: {
        nodes: featuredOrRelatedPosts
    }
};

export const mockGlossaryPageData = mockPageProps(
    mockGlossary
) as PageProps<GlossaryPageProps>;

export const mockGlossaryPageHeadData = mockHeadProps(
    mockGlossary
) as HeadProps<GlossaryPageProps>;
