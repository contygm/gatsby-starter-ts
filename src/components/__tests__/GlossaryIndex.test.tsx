import React from 'react';
import { render } from '@testing-library/react';
import { GlossaryIndex } from '../GlossaryIndex';
import { glossaryDefinitions } from '../../../__mocks__/constants';
import { mockGlossaryPageData } from '../../../__mocks__/mock-glossary-page';

describe('GlossaryIndex', () => {
    it('default renders correctly', () => {
        const { asFragment, getAllByTestId } = render(
            <GlossaryIndex allDefinitions={glossaryDefinitions} allLetters={mockGlossaryPageData.data.allLetters}/>
        );
        expect(asFragment()).toMatchSnapshot();
        expect(getAllByTestId('definition-card').length).toEqual(5);
    });

    it('default renders correctly with no results', () => {
        const { asFragment, getAllByTestId } = render(
            <GlossaryIndex allDefinitions={[]} />
        );
        expect(asFragment()).toMatchSnapshot();
        try {
            /**
             * NOTE: This should fail because allDefinitions is an empty array,
             * so there is no instance of a definition-card. Jest doesn't have a
             * built-in way to fail gracefully, so we have to use a try catch.
             *
             * If there is a definition-card for some reason, this test will still fail,
             * because in this try, we export().toBeUndefined().
             */
            expect(getAllByTestId('definition-card')).toBeUndefined();
        } catch (e) {
            // This should succeed, as we expect an undefined error
            expect(e.message).toContain('Unable to find an element by');
        }
    });
});
