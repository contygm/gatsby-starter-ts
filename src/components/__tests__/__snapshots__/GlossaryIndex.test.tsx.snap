// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlossaryIndex default renders correctly 1`] = `
<DocumentFragment>
  <article
    class="section"
  >
    <section
      class="container"
    >
      <div
        class="is-centered"
      >
        <section
          class="section"
        >
          <h2
            class="title is-size-2"
            id="A"
          >
            A
          </h2>
          <div
            class=""
            data-testid="definition-card"
          >
            <div
              class="card"
            >
              <div
                class="card-content"
              >
                <div
                  class="media"
                >
                  <div
                    class="media-content"
                  >
                    <div
                      class="level"
                    >
                      <div
                        class="level-left"
                      >
                        <div
                          class="level-item"
                        >
                          <h3
                            class="title is-3"
                            id="air"
                          >
                            Air
                          </h3>
                        </div>
                      </div>
                      <div
                        class="level-right"
                      >
                        <div
                          class="level-item has-text-centered"
                        >
                          <div>
                            <p
                              class="heading"
                            >
                              Syllables
                            </p>
                            <p
                              class="subheading"
                            >
                              a·ban·don
                            </p>
                          </div>
                        </div>
                        <div
                          class="level-item has-text-centered"
                        >
                          <div>
                            <p
                              class="heading"
                            >
                              Phonetics
                            </p>
                            <p
                              class="subheading"
                            >
                              əˈbandən
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="content"
                >
                  <div
                    class="container"
                  >
                    <h3>
                      Definition
                    </h3>
                    <p>
                      Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.
                    </p>
                  </div>
                </div>
                <div
                  class="content"
                >
                  <h4>
                    Similar Words: 
                  </h4>
                  <ol>
                    <li>
                      <a
                        class=""
                        href="#/blog/test"
                      >
                        /blog/test
                      </a>
                    </li>
                    <li>
                      <a
                        class=""
                        href="#/wiki/test"
                      >
                        /wiki/test
                      </a>
                    </li>
                  </ol>
                </div>
              </div>
              <div
                class="card-footer"
              >
                <p
                  class="mt-2 ml-2 mr-2"
                >
                  Tags:
                </p>
                <div
                  class="tags"
                >
                  <a
                    class="tag mt-2 is-success is-light"
                    href="?tag=two"
                    id="two"
                  >
                    two
                  </a>
                  <a
                    class="tag mt-2 is-success is-light"
                    href="?tag=three"
                    id="three"
                  >
                    three
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class=""
            data-testid="definition-card"
          >
            <div
              class="card"
            >
              <div
                class="card-content"
              >
                <div
                  class="media"
                >
                  <div
                    class="media-content"
                  >
                    <div
                      class="level"
                    >
                      <div
                        class="level-left"
                      >
                        <div
                          class="level-item"
                        >
                          <h3
                            class="title is-3"
                            id="atom"
                          >
                            Atom
                          </h3>
                        </div>
                      </div>
                      <div
                        class="level-right"
                      >
                        <div
                          class="level-item has-text-centered"
                        >
                          <div>
                            <p
                              class="heading"
                            >
                              Syllables
                            </p>
                            <p
                              class="subheading"
                            >
                              a·ban·don
                            </p>
                          </div>
                        </div>
                        <div
                          class="level-item has-text-centered"
                        >
                          <div>
                            <p
                              class="heading"
                            >
                              Phonetics
                            </p>
                            <p
                              class="subheading"
                            >
                              əˈbandən
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="content"
                >
                  <div
                    class="container"
                  >
                    <h3>
                      Definition
                    </h3>
                    <p>
                      Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.
                    </p>
                  </div>
                </div>
                <div
                  class="content"
                >
                  <h4>
                    Similar Words: 
                  </h4>
                  <ol>
                    <li>
                      <a
                        class=""
                        href="#/blog/test"
                      >
                        /blog/test
                      </a>
                    </li>
                    <li>
                      <a
                        class=""
                        href="#/wiki/test"
                      >
                        /wiki/test
                      </a>
                    </li>
                  </ol>
                </div>
              </div>
              <div
                class="card-footer"
              >
                <p
                  class="mt-2 ml-2 mr-2"
                >
                  Tags:
                </p>
                <div
                  class="tags"
                >
                  <a
                    class="tag mt-2 is-success is-light"
                    href="?tag=two"
                    id="two"
                  >
                    two
                  </a>
                  <a
                    class="tag mt-2 is-success is-light"
                    href="?tag=one"
                    id="one"
                  >
                    one
                  </a>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section
          class="section"
        >
          <h2
            class="title is-size-2"
            id="C"
          >
            C
          </h2>
          <div
            class=""
            data-testid="definition-card"
          >
            <div
              class="card"
            >
              <div
                class="card-content"
              >
                <div
                  class="media"
                >
                  <div
                    class="media-content"
                  >
                    <div
                      class="level"
                    >
                      <div
                        class="level-left"
                      >
                        <div
                          class="level-item"
                        >
                          <h3
                            class="title is-3"
                            id="cloud"
                          >
                            Cloud
                          </h3>
                        </div>
                      </div>
                      <div
                        class="level-right"
                      >
                        <div
                          class="level-item has-text-centered"
                        >
                          <div>
                            <p
                              class="heading"
                            >
                              Syllables
                            </p>
                            <p
                              class="subheading"
                            >
                              a·ban·don
                            </p>
                          </div>
                        </div>
                        <div
                          class="level-item has-text-centered"
                        >
                          <div>
                            <p
                              class="heading"
                            >
                              Phonetics
                            </p>
                            <p
                              class="subheading"
                            >
                              əˈbandən
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="content"
                >
                  <div
                    class="container"
                  >
                    <h3>
                      Definition
                    </h3>
                    <p>
                      Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.
                    </p>
                  </div>
                </div>
                <div
                  class="content"
                >
                  <h4>
                    Similar Words: 
                  </h4>
                  <ol>
                    <li>
                      <a
                        class=""
                        href="#/blog/test"
                      >
                        /blog/test
                      </a>
                    </li>
                    <li>
                      <a
                        class=""
                        href="#/wiki/test"
                      >
                        /wiki/test
                      </a>
                    </li>
                  </ol>
                </div>
              </div>
              <div
                class="card-footer"
              >
                <p
                  class="mt-2 ml-2 mr-2"
                >
                  Tags:
                </p>
                <div
                  class="tags"
                >
                  <a
                    class="tag mt-2 is-success is-light"
                    href="?tag=two"
                    id="two"
                  >
                    two
                  </a>
                  <a
                    class="tag mt-2 is-success is-light"
                    href="?tag=one"
                    id="one"
                  >
                    one
                  </a>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section
          class="section"
        >
          <h2
            class="title is-size-2"
            id="M"
          >
            M
          </h2>
          <div
            class=""
            data-testid="definition-card"
          >
            <div
              class="card"
            >
              <div
                class="card-content"
              >
                <div
                  class="media"
                >
                  <div
                    class="media-content"
                  >
                    <div
                      class="level"
                    >
                      <div
                        class="level-left"
                      >
                        <div
                          class="level-item"
                        >
                          <h3
                            class="title is-3"
                            id="moon"
                          >
                            Moon
                          </h3>
                        </div>
                      </div>
                      <div
                        class="level-right"
                      >
                        <div
                          class="level-item has-text-centered"
                        >
                          <div>
                            <p
                              class="heading"
                            >
                              Syllables
                            </p>
                            <p
                              class="subheading"
                            >
                              a·ban·don
                            </p>
                          </div>
                        </div>
                        <div
                          class="level-item has-text-centered"
                        >
                          <div>
                            <p
                              class="heading"
                            >
                              Phonetics
                            </p>
                            <p
                              class="subheading"
                            >
                              əˈbandən
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="content"
                >
                  <div
                    class="container"
                  >
                    <h3>
                      Definition
                    </h3>
                    <p>
                      Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.
                    </p>
                  </div>
                </div>
                <div
                  class="content"
                >
                  <h4>
                    Similar Words: 
                  </h4>
                  <ol>
                    <li>
                      <a
                        class=""
                        href="#/blog/test"
                      >
                        /blog/test
                      </a>
                    </li>
                    <li>
                      <a
                        class=""
                        href="#/wiki/test"
                      >
                        /wiki/test
                      </a>
                    </li>
                  </ol>
                </div>
              </div>
              <div
                class="card-footer"
              >
                <p
                  class="mt-2 ml-2 mr-2"
                >
                  Tags:
                </p>
                <div
                  class="tags"
                >
                  <a
                    class="tag mt-2 is-success is-light"
                    href="?tag=two"
                    id="two"
                  >
                    two
                  </a>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section
          class="section"
        >
          <h2
            class="title is-size-2"
            id="W"
          >
            W
          </h2>
          <div
            class=""
            data-testid="definition-card"
          >
            <div
              class="card"
            >
              <div
                class="card-content"
              >
                <div
                  class="media"
                >
                  <div
                    class="media-content"
                  >
                    <div
                      class="level"
                    >
                      <div
                        class="level-left"
                      >
                        <div
                          class="level-item"
                        >
                          <h3
                            class="title is-3"
                            id="window"
                          >
                            Window
                          </h3>
                        </div>
                      </div>
                      <div
                        class="level-right"
                      >
                        <div
                          class="level-item has-text-centered"
                        >
                          <div>
                            <p
                              class="heading"
                            >
                              Syllables
                            </p>
                            <p
                              class="subheading"
                            >
                              a·ban·don
                            </p>
                          </div>
                        </div>
                        <div
                          class="level-item has-text-centered"
                        >
                          <div>
                            <p
                              class="heading"
                            >
                              Phonetics
                            </p>
                            <p
                              class="subheading"
                            >
                              əˈbandən
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="content"
                >
                  <div
                    class="container"
                  >
                    <h3>
                      Definition
                    </h3>
                    <p>
                      Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.
                    </p>
                  </div>
                </div>
                <div
                  class="content"
                >
                  <h4>
                    Similar Words: 
                  </h4>
                  <ol>
                    <li>
                      <a
                        class=""
                        href="#/blog/test"
                      >
                        /blog/test
                      </a>
                    </li>
                    <li>
                      <a
                        class=""
                        href="#/wiki/test1"
                      >
                        /wiki/test1
                      </a>
                    </li>
                  </ol>
                </div>
              </div>
              <div
                class="card-footer"
              >
                <p
                  class="mt-2 ml-2 mr-2"
                >
                  Tags:
                </p>
                <div
                  class="tags"
                >
                  <a
                    class="tag mt-2 is-success is-light"
                    href="?tag=three"
                    id="three"
                  >
                    three
                  </a>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </section>
  </article>
</DocumentFragment>
`;

exports[`GlossaryIndex default renders correctly with no results 1`] = `
<DocumentFragment>
  <article
    class="section"
  >
    <section
      class="container"
    >
      <div
        class="is-centered"
      >
        <div>
          No results
        </div>
      </div>
    </section>
  </article>
</DocumentFragment>
`;
